### ISSUE TITLE
Bash Bundler TODO

### ISSUE THREAD

**Issue by shawnrice** [shawnrice/alfred-bundler#31]
Besides coding standards (#16), the Bash bundler needs a full rewrite.
#### Bundlet
- [x] ~~Coding Standards~~
- [x] ~~Install function~~
- [x] ~~Wrapper functions~~
  (Actually, I just cleaned out every function so that each does nothing, but all are overwritten by the backend).
#### Bundler
- [x] Coding standards
- [x] Icon
  - [x] System Icons
  - [x] Generic icons
  - [x] Color caching
  - [x] Rewrite color functions
- [x] Log function
  - [x] console logging
  - [x] file logging
  - [x] internal log
  - [x] user-facing log
- [x] ~~Load~~
- [x] ~~Utility~~
- [x] Implement new error codes #42 
#### Testing
- [x] alfred.bundler.sh
- [x] AlfredBundler.sh
- [x] icons.sh
- [x] logging.sh
- [x] math.sh

**Comment by shawnrice** [comment]
Icon fallbacks in place.

The only functionality left for the icon functions is to go through the icon servers in the file.

**Comment by shawnrice** [comment]
Added in fallback support for icon servers. We now read them out of the `icon_servers` file.

**Comment by shawnrice** [comment]
(#45) Added color caching for icons

**Comment by shawnrice** [comment]
Added reporting function (echoes to STDERR)
Started internal logging function (outputs to log files)

**Comment by shawnrice** [comment]
With 37b56a469de73175ef56d594e14a9cba8822297d, the math functions now have test coverage.

**Comment by shawnrice** [comment]
38546e7d7605c49ee82e8dcd736cce4d9c40472e moves the call to `AlfredBundler::bootstrap` inside the `AlfredBundler::main`. The reason for this is that the bootstrap function should run on call _only_ when the bundlet is called as a script rather than included as a library. So, for people using the bundlet as a library (i.e. workflows written in bash), they will be responsible for bootstrapping the bundler and checking for the exception.

**Comment by shawnrice** [comment]
And... I moved it back out of the function and removed the command to include the main bundler in the function. Instead, I've moved the bootstrap check into the actual file again. The problem was that the variables that needed to be declared in `AlfredBundler.sh` weren't being declared globally. Right now, the 'bootstrap' function isn't really working correctly as an exception handler. Working on it.
